test/test_pipeline/components/data_preprocessing/test_data_preprocessing_text.py

```diff
@@ -0,0 +1,65 @@
+import unittest
+import numpy as np
+import pandas as pd
+from scipy import sparse
+
+import pytest
+
+from autosklearn.pipeline.components.data_preprocessing.text_encoding.bag_of_word_encoding import \
+    BagOfWordEncoder as BOW
+from autosklearn.pipeline.components.data_preprocessing.text_encoding.bag_of_word_encoding_distinct import \
+    BagOfWordEncoder as BOW_distinct
+
+
+class CategoricalPreprocessingPipelineTest(unittest.TestCase):
+
+    def test_fit_transform(self):
+        X = pd.DataFrame({"col1": ["Hello World",
+                                   "This is a test"],
+                          "col2": ["Hello Mars",
+                                   "This is the second column"]}).astype({"col1": "string",
+                                                                          "col2": "string"})
+        BOW_fitted = BOW(ngram_range=1, min_df_choice="min_df_absolute", min_df_absolute=0,
+                         min_df_relative=0, random_state=1).fit(X.copy())
+
+        Yt = BOW_fitted.preprocessor.vocabulary_
+        words = sorted(["hello", "world", "this", "is", "test",  # "a" is not added, len(...)=1
+                        "mars", "the", "second", "column"])      # is ignored by CountVectorizer
+        Y = {key: idx for idx, key in enumerate(words)}
+
+        np.testing.assert_array_equal(Yt, Y)
+
+        BOW_fitted = BOW_distinct(ngram_range=1, min_df_choice="min_df_absolute", min_df_absolute=0,
+                                  min_df_relative=0, random_state=1).fit(X.copy())
+
+        for key in BOW_fitted.preprocessor:
+            y = []
+            for col in X[key]:
+                y += [word for word in col.lower().split(" ") if len(word) > 1]
+            y = sorted(y)
+            yt = sorted(BOW_fitted.preprocessor[key].vocabulary_.keys())
+            np.testing.assert_array_equal(yt, y)
```

Comment:
No need to change it now, just in the future, this test should be split into two as it's testing two different things.